modin/core/execution/dask/common/utils.py

```diff
@@ -50,7 +50,7 @@ def _disable_warnings():
             """
     from distributed import Client
 
-    client = Client()
+    client = Client(n_workers=<N1>, threads_per_worker=<N2>)
```

Comment:
Don't you think this is an additional complication for users?

Let's also discuss warning in general.

Don't you think that such a warning encourages users to initialize the engine themselves (a common reaction to avoid annoying warnings), which in many situations is worse than what Modin does itself.

```python
>>> import modin.pandas as pd
>>> pd.DataFrame([1,2,3,4]) 
UserWarning: Ray execution environment not yet initialized. Initializing...
To remove this warning, run the following python code before doing dataframe operations:

    import ray
    ray.init()

2024-02-04 16:18:03,686 INFO worker.py:1673 -- Started a local Ray instance.
UserWarning: Distributing <class 'list'> object. This may take some time.
   0
0  1
1  2
2  3
3  4
```

Reply:
This might look like extra burden for the user but we intentionally emit this kind of a warning to emphasize that the engine can be initialized outiside Modin and it in turn can connent to the engine. I would rephrase the warning a bit. What do you think on this?

```bash
UserWarning: Ray execution environment not yet initialized. Initializing...
To remove this warning you may want to initialize the engine on your own
with the specific parameters passed in:

import ray

ray.init(num_cpus=<N>, ...)
```

Reply:
> the warning encourages users to initialize the engine themselves, which in many situations is worse than what Modin does itself

I also agree with this. For ray, it could be reflected with memory problems if a user would try to initialize the engine themself without any arguments:
```python
import ray
ray.init() # only allowed to use 30% of /dev/shm's memory

# -------------------------------
import modin.pandas as pd
pd.DataFrame(...) # allowed to use 60% of /dev/shm's memory
``` 
We could have added additional instructions to the warning on how to specify the proper amount of memory, but IMO that would make the warning unreadable and not understandable, as the whole process of specifying the memory is quite complicated:
```python
# in order to initialize Ray with the same amount of object's store memory
# as with Modin, there's too much to consider for a user IMO
import ray
# at first, I have to know that my system have 16gb of shared memory,
# and then I also have to do math homework
ray.init(object_store_memory=16 * 1024 * 1024 * 1024 * 0.6)
```

So, I'm siding here with @anmyachev and wondering whether we should keep the warning at all or it's better to remove it

Reply:
In order to put my question simpler:
1. What are the use cases where a user need to materialize the engine on their own? Modin have a lot of env variables that can control the arguments of the engine (`MODIN_CPUS`, `MODIN_MEMORY`, `MODIN_CLUSTER_PASSWORD`, ...), aren't they enough?
2. If cases where it's absolutely necessary to initialize the engine on your own are rather exceptions (as you can control pretty much everything via modin config), is it worth printing a 7-line warning all the time?

Reply:
> 1. What are the use cases where a user need to initialize the engine on their own? Modin have a lot of env variables that can control the arguments of the engine (`MODIN_CPUS`, `MODIN_MEMORY`, `MODIN_CLUSTER_PASSWORD`, ...), aren't they enough?

The use of the Modin configs might be a little more cumbersome as opposed to initializing the engine by users themselves. They may want to initailize the engine on their own with the specific configurations, which may also not exist in Modin.

> If cases where it's absolutely necessary to initialize the engine on your own are rather exceptions (as you can control pretty much everything via modin config), is it worth printing a 7-line warning all the time?

```bash
UserWarning: Ray execution environment not yet initialized. Initializing...
You may want to initialize Ray on your own with the specific configuration.
Refer to Ray documentation for this purpose on more details.
```

What do you think on this?

Reply:
I want to prevent users from initializing an engine with either `ray.init()` or `Client()`. Particularly, I am worried about Dask, which is much slower as said [earlier](https://github.com/modin-project/modin/pull/6909#discussion_r1477127503) with the default client. I think it would be good to emit a warning like this.

```bash
UserWarning: Ray execution environment not yet initialized. Initializing...
You may want to initialize Ray on your own with the specific configuration.
Refer to Ray documentation for this purpose on more details.
```

If the user is fair enough aware of the engine to be initialized, he/she can look at the engine docs and choose the necessary parameters for the initialization.

Reply:
> This looks better to me, compared with what we have now. However, I'm still skeptical regarding having this warning at all, since in 90%+ use cases you don't need to initialize on your own. Though, I'm not going to block the decision of keeping the warning, let's wait for @anmyachev opinion

Agree. 

> I want to prevent users from initializing an engine with either ray.init() or Client(). Particularly, I am worried about Dask, which is much slower as said https://github.com/modin-project/modin/pull/6909#discussion_r1477127503 with the default client. I think it would be good to emit a warning like this.

We can write about this in our documentation, in the section on advanced usage (and remove this warning). Also, instead of warning, we can do logging, for example, of the info level.

Reply:
I am fine by removing the warning. Will add a section in the docs.

Reply:
Done